extensions/sql-database-projects/src/dialogs/deployDatabaseDialog.ts

```diff
@@ -104,14 +104,59 @@ export class DeployDatabaseDialog {
 		});
 	}
 
-	private async deploy(): Promise<void> {
-		// TODO: hook up with build and deploy
+	private async getConnectionUri(): Promise<string> {
 		// if target connection is a data source, have to check if already connected or if connection dialog needs to be opened
+		let connId: string;
+
+		if (this.dataSourcesRadioButton?.checked) {
+			const dataSource = (this.dataSourcesDropDown!.value! as DataSourceDropdownValue).dataSource;
+
+			const connProfile: azdata.IConnectionProfile = {
+				serverName: dataSource.server,
+				databaseName: dataSource.database,
+				connectionName: dataSource.name,
+				userName: '',
+				password: '', // TODO: secure password storage
+				authenticationType: dataSource.integratedSecurity ? 'Integrated' : 'SqlAuth',
+				savePassword: false,
+				providerName: 'MSSQL',
+				saveProfile: true,
+				id: dataSource.name + '-dataSource',
+				options: []
+			};
+
+			const connResult = await azdata.connection.connect(connProfile, false, false);
+			connId = connResult.connectionId;
+		}
+		else {
+			const connection = (await azdata.connection.getConnections()).find(c => c.connectionName === this.targetConnectionTextBox!.value)!;
```

Comment:
Would this work if the connection isn't named? Also, the deployDatabaseDialog.connection is set to the selected connection when a connection is selected using the Edit button so that can be used instead of searching through the connections.

Reply:
Hmm... to both this comment and the one about SqlAuth, I think I'll need to some help defining the behavior here.  Ultimately, this method's output should be the Uri within azdata.connection that should be used for dacFxService.